src/scheduler/MpiWorld.cpp

```diff
@@ -305,46 +307,30 @@ int MpiWorld::isend(int sendRank,
                     int count,
                     faabric::MPIMessage::MPIMessageType messageType)
 {
-    return doISendRecv(
-      sendRank, recvRank, buffer, nullptr, dataType, count, messageType);
+    int requestId = (int)faabric::util::generateGid();
```

Comment:
As discussed, removed the `doISendRecv` method and implemented separate `irecv` and `isend`.